tensordict/nn/tensorclass_module.py

```diff
@@ -0,0 +1,242 @@
+    keys: List[Tuple[str, ...]] = []
+    for field in fields:
+        key = field.name
+        if issubclass(field.type, TensorClass):
+            subkeys = _tensor_class_keys(cast(type[TensorClass], field.type))
+            for subkey in subkeys:
+                keys.append((key,) + subkey)
+        else:
+            keys.append((key,))
+    return keys
+
+
+InputTensorClass = TypeVar("InputTensorClass", bound=TensorClass)
+OutputTensorClass = TypeVar("OutputTensorClass", bound=TensorClass)
+
+
+class TensorClassModuleWrapper(TensorDictModuleBase):
+    """Wrapper class for TensorClassModuleBase objects.
+
+    This wrapper allows TensorClassModuleBase instances to be used in TensorDict-based
+    workflows by handling the conversion between TensorDict and TensorClass representations.
+    When called with a TensorDict, the wrapper converts it to a TensorClass, passes it through
+    the wrapped module, and converts the output back to a TensorDict.
+
+    Args:
+        module (TensorClassModuleBase): The TensorClassModuleBase instance to wrap.
+
+    Examples:
+        >>> from tensordict import TensorDict
+        >>> from tensordict.tensorclass import TensorClass
+        >>> from tensordict.nn import TensorClassModuleBase
+        >>> import torch
+        >>>
+        >>> class InputTC(TensorClass):
+        ...     x: torch.Tensor
+        ...
+        >>> class OutputTC(TensorClass):
+        ...     y: torch.Tensor
+        ...
+        >>> class MyModule(TensorClassModuleBase[InputTC, OutputTC]):
+        ...     def forward(self, input: InputTC) -> OutputTC:
+        ...         return OutputTC(y=input.x + 1, batch_size=input.batch_size)
+        ...
+        >>> module = MyModule()
+        >>> td_module = module.as_td_module()
+        >>> td = TensorDict({"x": torch.zeros(3)}, batch_size=[3])
+        >>> result = td_module(td)
+        >>> assert "y" in result
+
+    """
+
+    def __init__(
+        self, module: TensorClassModuleBase[InputTensorClass, OutputTensorClass]
+    ) -> None:
+        super().__init__()
+        self.tc_module = module
+        self.in_keys = _tensor_class_keys(cast(type[TensorClass], module.input_type))
+        self.out_keys = _tensor_class_keys(cast(type[TensorClass], module.output_type))
+
+    @dispatch(auto_batch_size=False)
+    def forward(self, tensordict: TensorDict, *args, **kwargs) -> TensorDict:
+        """Forward pass converting TensorDict to TensorClass and back.
+
+        Args:
+            tensordict (TensorDict): Input tensordict.
+            *args: Additional positional arguments.
+            **kwargs: Additional keyword arguments.
+
+        Returns:
+            TensorDict: Output tensordict.
+
+        """
+        return self.tc_module(
+            self.tc_module.input_type.from_dict(
+                input_dict=tensordict.to_dict(),
+                batch_size=tensordict.batch_size,
+                device=tensordict.device,
+            )
```

Comment:
can't we use `TensorClass.from_tensordict` directly?
The issue I see here is for example that a tensorclass can contain a LazyStackedTensorDict with exclusive keys (keys present in some but not all elements of the stack). `to_dict` will fail for these. It's also slow.